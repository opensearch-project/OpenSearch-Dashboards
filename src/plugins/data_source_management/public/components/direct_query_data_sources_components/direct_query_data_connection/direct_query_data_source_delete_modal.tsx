/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from 'react';
import {
  EuiOverlayMask,
  EuiModal,
  EuiSmallButton,
  EuiSmallButtonEmpty,
  EuiCompressedFieldText,
  EuiForm,
  EuiCompressedFormRow,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiSpacer,
  EuiText,
} from '@elastic/eui';

export const DeleteModal = ({
  onCancel,
  onConfirm,
  title,
  message,
  prompt,
}: {
  onCancel: (
    event?: React.KeyboardEvent<HTMLDivElement> | React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void;
  onConfirm: (event?: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  title: string;
  message: string;
  prompt?: string;
}) => {
  const [value, setValue] = useState('');
  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };
  const deletePrompt = prompt ?? 'delete';
  return (
    <EuiOverlayMask>
      <EuiModal onClose={onCancel} initialFocus="[name=input]">
        <EuiModalHeader>
          <EuiModalHeaderTitle>{title}</EuiModalHeaderTitle>
        </EuiModalHeader>

        <EuiModalBody>
          <EuiText>{message}</EuiText>
          <EuiText>The action cannot be undone.</EuiText>
          <EuiSpacer />
          <EuiForm>
            <EuiCompressedFormRow
              label={`To confirm deletion, enter "${deletePrompt}" in the text field`}
            >
              <EuiCompressedFieldText
                name="input"
                placeholder={deletePrompt}
                value={value}
                onChange={(e) => onChange(e)}
                data-test-subj="popoverModal__deleteTextInput"
              />
            </EuiCompressedFormRow>
          </EuiForm>
        </EuiModalBody>

        <EuiModalFooter>
          <EuiSmallButtonEmpty onClick={onCancel}>Cancel</EuiSmallButtonEmpty>
          <EuiSmallButton
            onClick={() => onConfirm()}
            color="danger"
            fill
            disabled={value !== deletePrompt}
            data-test-subj="popoverModal__deleteButton"
          >
            Delete
          </EuiSmallButton>
        </EuiModalFooter>
      </EuiModal>
    </EuiOverlayMask>
  );
};
