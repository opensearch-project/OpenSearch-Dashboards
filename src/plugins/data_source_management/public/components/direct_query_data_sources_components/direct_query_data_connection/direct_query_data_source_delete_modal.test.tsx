/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import { DeleteModal } from './direct_query_data_source_delete_modal';
import {
  EuiButtonEmpty,
  EuiFieldText,
  EuiModalBody,
  EuiModalHeaderTitle,
  EuiText,
} from '@elastic/eui';

describe('DeleteModal', () => {
  const defaultProps = {
    onCancel: jest.fn(),
    onConfirm: jest.fn(),
    title: 'Delete Item',
    message: 'Are you sure you want to delete this item?',
    prompt: 'delete',
  };

  const shallowComponent = (props = defaultProps) => shallow(<DeleteModal {...props} />);
  const mountComponent = (props = defaultProps) => mount(<DeleteModal {...props} />);

  test('renders correctly', () => {
    const wrapper = shallowComponent();
    expect(wrapper).toMatchSnapshot();
  });

  test('displays the correct title and message', () => {
    const wrapper = mountComponent();
    expect(wrapper.find(EuiModalHeaderTitle).text()).toEqual('Delete Item');
    expect(wrapper.find(EuiModalBody).find(EuiText).at(0).text()).toEqual(
      'Are you sure you want to delete this item?'
    );
  });

  test('disables the delete button initially', () => {
    const wrapper = mountComponent();
    const deleteButton = wrapper.find('button[data-test-subj="popoverModal__deleteButton"]');
    expect(deleteButton.prop('disabled')).toBe(true);
  });

  test('enables the delete button when the correct prompt is entered', () => {
    const wrapper = mountComponent();
    const input = wrapper.find(EuiFieldText);
    input.simulate('change', { target: { value: 'delete' } });
    wrapper.update();
    const deleteButton = wrapper.find('button[data-test-subj="popoverModal__deleteButton"]');
    setTimeout(() => {
      expect(deleteButton.prop('disabled')).toBe(false);
    }, 1000);
  });

  test('calls onCancel when the cancel button is clicked', () => {
    const wrapper = mountComponent();
    wrapper.find(EuiButtonEmpty).simulate('click');
    expect(defaultProps.onCancel).toHaveBeenCalled();
  });

  test('calls onConfirm when the delete button is clicked', () => {
    const wrapper = mountComponent();
    const input = wrapper.find(EuiFieldText);
    input.simulate('change', { target: { value: 'delete' } });
    wrapper.update();
    const deleteButton = wrapper.find('button[data-test-subj="popoverModal__deleteButton"]');
    setTimeout(() => {
      deleteButton.simulate('click');
      expect(defaultProps.onConfirm).toHaveBeenCalled();
    }, 1000);
  });
});
