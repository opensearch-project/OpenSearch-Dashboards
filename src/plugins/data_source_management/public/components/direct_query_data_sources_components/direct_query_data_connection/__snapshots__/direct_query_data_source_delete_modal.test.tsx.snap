// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal renders correctly 1`] = `
<EuiOverlayMask>
  <EuiModal
    initialFocus="[name=input]"
    onClose={[MockFunction]}
  >
    <EuiModalHeader>
      <EuiModalHeaderTitle>
        Delete Item
      </EuiModalHeaderTitle>
    </EuiModalHeader>
    <EuiModalBody>
      <EuiText>
        Are you sure you want to delete this item?
      </EuiText>
      <EuiText>
        The action cannot be undone.
      </EuiText>
      <EuiSpacer />
      <EuiForm>
        <EuiCompressedFormRow
          describedByIds={Array []}
          display="rowCompressed"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          label="To confirm deletion, enter \\"delete\\" in the text field"
          labelType="label"
        >
          <EuiCompressedFieldText
            data-test-subj="popoverModal__deleteTextInput"
            name="input"
            onChange={[Function]}
            placeholder="delete"
            value=""
          />
        </EuiCompressedFormRow>
      </EuiForm>
    </EuiModalBody>
    <EuiModalFooter>
      <EuiSmallButtonEmpty
        onClick={[MockFunction]}
      >
        Cancel
      </EuiSmallButtonEmpty>
      <EuiSmallButton
        color="danger"
        data-test-subj="popoverModal__deleteButton"
        disabled={true}
        fill={true}
        onClick={[Function]}
      >
        Delete
      </EuiSmallButton>
    </EuiModalFooter>
  </EuiModal>
</EuiOverlayMask>
`;
